services/servicegroup/service/src/main/java/org/collectionspace/services/servicegroup/nuxeo/ServiceGroupDocumentModelHandler.java

```diff
@@ -162,79 +160,138 @@ public PoxPayloadOut getResourceItemForCsid(ServiceContext<PoxPayloadIn, PoxPayl
+		List<ServiceBindingType> servicebindings =
+			tReader.getServiceBindingsByType(ctx.getTenantId(), serviceGroupNames);
+		if (servicebindings == null || servicebindings.isEmpty()) {
+			Response response = Response.status(Response.Status.NOT_FOUND)
+				.entity(ServiceMessages.READ_FAILED +
+						ServiceMessages.resourceNotFoundMsg(implode(serviceGroupNames, ",")))
+				.type(MediaType.TEXT_PLAIN_TYPE).build();
+			throw new CSWebApplicationException(response);
+		}
+
+		String queryTag = ctx.getQueryParams().getFirst(IQueryManager.TAG_QUERY_PARAM);
+		servicebindings = SecurityUtils.getReadableServiceBindingsForCurrentUser(servicebindings);
+		// Build the list of docTypes for allowed serviceBindings filtered optionally on tags
+		ArrayList<String> docTypes = new ArrayList<String>();
+		for (ServiceBindingType binding : servicebindings) {
+			boolean acceptDocType = true;
+			if (queryTag != null && !queryTag.isEmpty()) {
+				acceptDocType = acceptServiceBinding(binding, queryTag);
+			}
+
+			ServiceObjectType serviceObj = binding.getObject();
+			if (acceptDocType && serviceObj != null) {
+				String docType = serviceObj.getName();
+				docTypes.add(docType);
+				queriedServiceBindings.put(docType, binding);
+			}
+		}
+
+		// we can't query if no doc types match, likely from a bad tag
+		if (docTypes.isEmpty()) {
+			Response response = Response.status(Response.Status.BAD_REQUEST)
+				.entity("Cannot query against no document types. Check that tag query parameters are correct.")
+				.type(MediaType.TEXT_PLAIN_TYPE)
+				.build();
+			throw new CSWebApplicationException(response);
+		}
+
+		// This should be type "Document" but CMIS is gagging on that right now.
+		ctx.getQueryParams().add(
+			IQueryManager.SELECT_DOC_TYPE_FIELD,
+			QueryManagerNuxeoImpl.COLLECTIONSPACE_DOCUMENT_TYPE);
+
+		// Now we have to issue the search
+		// The findDocs() method will build a QueryContext, which wants to see a docType for our context
+		ctx.setDocumentType(QueryManagerNuxeoImpl.NUXEO_DOCUMENT_TYPE);
+		DocumentWrapper<DocumentModelList> docListWrapper = repoClient.findDocs(ctx, this, repoSession, docTypes);
+		// Now we gather the info for each document into the list and return
+		DocumentModelList docList = docListWrapper.getWrappedObject();
+		return docList;
+	}
+
+	/**
+	 * Test if a service binding should be included in a search based on the tag query parameter passed in.
+	 * <p>
+	 * If a tag starts with a "-", it is assumed this is intended to be a negation, and we check if the service binding
+	 * does not contain said tag.
+	 *
+	 * @param binding the ServiceBinding to check
+	 * @param queryTag tag query parameter
+	 * @return true if the ServiceBinding contains all query parameters, false otherwise
+	 */
+	public boolean acceptServiceBinding(ServiceBindingType binding, String queryTag) {
+		final Tags tags = binding.getTags();
+		if (tags == null || tags.getTag().isEmpty()) {
+			return false;
```

Comment:
Can't return false here now that we have negation. You probably just want to initialize null to an empty list, and let it fall through.

Reply:
Oh yep, good catch. I'll update this and get a test in for it too.